artemis-protocols/artemis-openwire-protocol/src/main/java/org/apache/activemq/artemis/core/protocol/openwire/OpenWireMessageConverter.java

```diff
@@ -610,7 +613,7 @@ private static ActiveMQMessage toAMQMessage(MessageReference reference,
          amqMsg.setCorrelationId(corrId.toString());
       }
 
-      final byte[] dsBytes = (byte[]) coreMessage.getObjectProperty(AMQ_MSG_DATASTRUCTURE);
+      final byte[] dsBytes = getBytesPropertyOrNull(coreMessage, AMQ_MSG_DATASTRUCTURE);
```

Comment:
This would been either a mistake from the user, or someone maliciously trying to break things. 

shouldn't you log such condition instead of silently failing?

or perhaps I would prefer rejecting the message and sending an error to the client, meaning the previous semantic was probably ok?

Reply:
I thought about this, but this actually came out from a legitimate use case as described in the Jira:
1. Send a message to the broker using OpenWire client
2. Consume that message using AMQP client
3. Send that message back to the broker using AMQP client

Considering the previous use case, at which level you would log?

Reply:
Definitely log.warn

Reply:
>>Send a message to the broker using OpenWire client

Perhaps we could fix whatever conversion made that happen?

Reply:
> > > Send a message to the broker using OpenWire client
> 
> Perhaps we could fix whatever conversion made that happen?

Agree. I think the problem is located in the AMQP module but it needs more investigation. Can we create another Jira for that and release this fix?

Reply:
> Hi @jbertram @clebertsuconic , what about merging this?

reading others reviews, isn't this waiting of finding the underlying cause why the properties go missing in AMQP? we should be fixing underlying issue, not plastering over.